### DISCUSSION TITLE
Issues with Saving Project When Refreshing Page?

### DISCUSSION THREAD

**Discussion by gauravsingh94** [processing/p5.js-web-editor#2793] [General]
### Increasing Access

**Existing Problem**

1. When we are refreshing the accedently remove the ide content.

2 Storing what they have written to the local store would be benifical they will not lost their work if they accidently refresh their page.

### Feature enhancement details

3. i will implement a buttom same as auto refresh named "auto save" it will the project automatically.

Please give me your views on this so that i can start working on it if it feels legit to you.

**Comment by gauravsingh94** [comment]
@lindapaiste 
Please give me your views on this so that i can start working on it if it feels legit to you.

**Comment by SasiDharan731** [comment]
This seems like a good idea and I think this issue should be moved into the newly introduced discussion tab. @raclim

**Comment by raclim** [comment]
We do already have 'Autosave' in the Settings modal and I think we should already be informing users if they're closing out a page with unsaved changes, but I could be wrong on that one. I agree and think we could see if others also face these types of saving issues and how they currently handle it, or what some other workarounds could be!

**Reply by lindapaiste** [reply]
We're *supposed* to have that alert but we don't right now and that's the problem. 

Explained here: https://github.com/processing/p5.js-web-editor/pull/2590#issuecomment-1806901605

I should do this because I'm the only one who understands it 😆.

Not sure if you want to reactivate the issue or open a new one?

**Comment by gauravsingh94** [comment]
> We do already have 'Autosave' in the Settings modal and I think we should already be informing users if they're closing out a page with unsaved changes, but I could be wrong on that one. I agree and think we could see if others also face these types of saving issues and how they currently handle it, or what some other workarounds could be!

The resolution to this issue involves

If the changes aren't saved and the page is refreshed, a confirmation message should appear, alerting the user that their modifications have not been saved and will be lost upon refresh.

 For instance, a message similar to "Your changes have not been saved and will be lost" can be displayed in such cases.
